### DISCUSSION TITLE
日常 action 已接近六小时

### DISCUSSION THREAD

**Discussion by PirateBook1** [banned-historical-archives/banned-historical-archives.github.io#2280] [General]
@banned-historical-archives 
现在的静态页面构建不是增量构建吗，action 已经接近六小时了，数据库很快就要无法加入数据了

**Comment by banned-historical-archives** [comment]
计算tags的时间有点长，可能也要像ocr一样加缓存

**Reply by PirateBook1** [reply]
说到标签这个事，机器计算+校对者最后审定可能是广义作者的一种解决方案。

**Reply by PirateBook1** [reply]
@banned-historical-archives 把 tags 做一个缓存，然后校对者可以去改这个 tags，基本上就是当时讨论的“广义作者”的解决方案了

**Reply by banned-historical-archives** [reply]
如果有特殊情况，tags可以在classifier里面直接修改。这和缓存没有关系。

**Reply by PirateBook1** [reply]
不是，我的意思是直接针对现有的标签把这个功能给实现了

**Reply by PirateBook1** [reply]
最好是在一篇文稿的校对结束后，校对者一并将标签给敲定，目前程序直接匹配出来的标签完全没法用，一些重要人物每篇都提到，但文里讲的内容不一定是跟这个人有关

**Reply by banned-historical-archives** [reply]
人工写标签很容易遗漏

**Reply by PirateBook1** [reply]
把程序匹配出来的标签交给人工最后确定。不是遗漏与不遗漏的问题，问题在于把“标签”这个功能给利用上，没有必要文中提到一次“林彪”就把“林彪”加在标签里面。这个东西从定位上就是一个很主观的功能，也无所谓什么遗漏与不遗漏了

**Reply by PirateBook1** [reply]
https://github.com/banned-historical-archives/banned-historical-archives.github.io/discussions/831#discussioncomment-4003504

**Reply by PirateBook1** [reply]
根据开发事项的轻重缓急来看吧

**Comment by banned-historical-archives** [comment]
主要消耗在生成html

**Reply by banned-historical-archives** [reply]
如果放弃seo可以快很多 @PirateBook1

**Reply by PirateBook1** [reply]
每次运行都一定要重新生成一遍全部吗

**Reply by banned-historical-archives** [reply]
也可以本地运行，然后推到指定分支上

**Reply by banned-historical-archives** [reply]
本地构建

**Reply by PirateBook1** [reply]
想办法把运行时间给降下来

**Comment by PirateBook1** [comment]
@banned-historical-archives 时间还没有降下来

**Reply by banned-historical-archives** [reply]
比较难搞

**Reply by PirateBook1** [reply]
把 tags 提出来缓存吧

**Reply by banned-historical-archives** [reply]
tags缓存作用不大，最多节约10分钟

**Reply by banned-historical-archives** [reply]
我大概知道什么原因了

**Reply by PirateBook1** [reply]
@banned-historical-archives 可以修复吗

**Reply by banned-historical-archives** [reply]
正在修

**Comment by PirateBook1** [comment]
@banned-historical-archives 构建有时候会失败，这是什么原因

**Reply by banned-historical-archives** [reply]
Error: The process '/usr/bin/git' failed with exit code 1
这种错误应该是两个commit间隔太小了，发布的时候会冲突，其中一个会被忽略。
